test/unit/lib/meli-client.test.js

```diff
@@ -11,7 +11,8 @@ const Account = require('../../../model/account')
 // Mount db connection
 require('../../../config/db').connect()
 
-const devAccountUsername = process.env.DEV_ACCOUNT_USERNAME
+const testAccountUsernames = process.env.TEST_ACCOUNT_USERNAMES && process.env.TEST_ACCOUNT_USERNAMES.split(',')
+const devAccountUsername = process.env.DEV_ACCOUNT_USERNAME || testAccountUsernames[0]
```

Comment:
Tests should be self-contained and they must not depend on the environment to have precision, consistency and be tested cross platform, this users must be on a test builder/fixture and database should be in-memory o docker coded-dependant

Reply:
You have a point here... But these tests are actually more 'integration' than 'unit'. The main reason of having a persistent DB in this case, is keeping the meli accounts access tokens up to date (since we are doing real requests using MeliClient). To have all self-contained as you say, I'd have to stub the meli client as well and use mock responses. Besides the extra work of doing this: considering that the Meli API changes regularly and some bugs can arise and go unnoticed...  is it a good idea to take this path?

Reply:
It's ok to leave production/test MeliAPI enviroment on your tests because is a project that depends on it and you are going to be noticed when tests are failing cause meli changed the approach but you shouldn't be using OS installed mongo, because for example, if you want to run this tests on a CI or on another system you'll have to take extra steps. If those are integration/e2e tests instead of unit ones you should use a docker mongo, setup it before tests and run the tests.

Reply:
Keep in mind that the tests are running correctly in TravisCI right now :) The mongo DB instance doesn't have to be OS dependant as it's configurable in MONGODB_URL env var. The URL can be a localhost port or a remote URL such as a mlab server or something else.